jobs/payment-jobs/tasks/routing_slip_task.py

```diff
@@ -333,6 +345,98 @@ def _get_routing_slip_by_status(cls, status: RoutingSlipStatus) -> list[RoutingS
             .all()
         )
 
+    @classmethod
+    def _get_applied_invoices_amount(cls, routing_slip: RoutingSlipModel) -> float:
+        """Get total amount of applied invoices in SBC-PAY for routing slip."""
+        total_applied = (
+            db.session.query(func.sum(InvoiceModel.total))
+            .filter(
+                InvoiceModel.routing_slip == routing_slip.number,
+                InvoiceModel.invoice_status_code == InvoiceStatus.PAID.value
+            )
+            .scalar()
+        ) or 0.0
+
+        return float(total_applied)
+
+    @classmethod
+    def _check_data_consistency(
+        cls,
+        routing_slip: RoutingSlipModel,
+        sbc_pay_applied_amount: float,
+        cfs_receipt_details: list[dict]
+    ) -> None:
+        """Check data consistency between SBC-PAY and CFS."""
+        sbc_pay_has_invoices = sbc_pay_applied_amount > 0
+        cfs_has_invoices = any(
+            receipt['has_applied_invoices'] 
+            for receipt in cfs_receipt_details
+        )
+
+        if sbc_pay_has_invoices and not cfs_has_invoices:
+            error_msg = (
+                f"Data mismatch for routing slip {routing_slip.number}: "
+                f"SBC-PAY shows ${sbc_pay_applied_amount:.2f} in applied "
+                f"invoices, but CFS shows no applied invoices. "
+                f"Manual intervention required to apply invoices in CFS."
+            )
+            if not current_app.config.get("DISABLE_RS_ADJUSTMENT_ERROR_EMAIL"):
+                notification = JobFailureNotification(
+                    subject="Routing Slip Adjustment Job Failure",
+                    file_name="routing_slip_adjustment",
+                    error_messages=[{"error": error_msg}],
+                    table_name=None,
+                    job_name="Routing Slip Adjustment Job",
+                )
+            notification.send_notification()
+            raise ValueError(error_msg)
```

Comment:
worth putting in a column under `routing_slips`  

`cas_mismatch` column?

that way you could just exclude those ones, also we could use `cas_mismatch` for reconciliation?